xbmc/Application.cpp

```diff
@@ -1123,8 +1124,22 @@ bool CApplication::Initialize()
   g_curlInterface.Unload();
 
   // initialize (and update as needed) our databases
-  CSplash::GetInstance().Show(g_localizeStrings.Get(24150));
-  CDatabaseManager::GetInstance().Initialize();
+  std::atomic<bool> notFinished = true;
+  CJobManager::GetInstance().Submit([&notFinished]() {
+    CDatabaseManager::GetInstance().Initialize();
+    notFinished = false;
+  });
+  std::string localizedStr = g_localizeStrings.Get(24150);
+  while (notFinished)
+  {
+    for (int i = 1; i <= 3 && notFinished; ++i)
+    {
+      if (CDatabaseManager::GetInstance().m_bIsUpgrading)
```

Comment:
notFinished is about the whole procedure. Not only the actual upgrade but also checking if there iis an old db and copying if we don't find one.  Isupgrading is when the actual migration from an old one to a new one happens. That's the slow part we want to display.
There are 2 different because there's no simple check ahead of time if we need to do some upgrade. For every database we get back version by version to find if there is a ddatabase.  If the upgrade fails we keep going back..